etl/src/main/java/org/pih/hivmigration/etl/sql/LabResultMigrator.groovy

```diff
@@ -0,0 +1,193 @@
+                     r.sample_id, 
+                     lower(t.name) as test_type,
+                     to_char(e.encounter_date, 'yyyy-mm-dd') as obs_datetime,
+                     value as value_numeric, 
+                     value_string as value_text,   
+                     case when (value_p='t') then ( 1)     
+                        else 0
+                        end as value_boolean,                
+                     case when (r.RESULT_UNDETECTABLE='t') then ( 1)     
+                        else 0
+                        end as vl_beyond_detectable_limit,                  
+                     r.RESULT_UNDETECTABLE_RANGE as vl_detectable_lower_limit
+            from     hiv_demographics_real d, hiv_encounters e, hiv_lab_results r, hiv_lab_tests t
+            where    d.patient_id = e.patient_id
+            and      e.encounter_id = r.encounter_id
+            and      r.lab_test_id = t.lab_test_id  
+            order by d.patient_id desc, e.encounter_id desc
+        ''')
+
+        executeMysql("Add UUIDs", "UPDATE hivmigration_lab_results SET uuid = uuid();")
+
+        executeMysql("Create stored procedures to add obs", '''
+            DROP PROCEDURE IF EXISTS create_tmp_obs_table;
+            DROP PROCEDURE IF EXISTS migrate_tmp_obs;
+            DELIMITER $$ ;
+            CREATE PROCEDURE create_tmp_obs_table()
+            BEGIN
+                DROP TABLE IF EXISTS tmp_obs;
+                CREATE TABLE tmp_obs (
+                    obs_id INT PRIMARY KEY AUTO_INCREMENT,
+                    obs_group_id INT,
+                    source_patient_id INT,
+                    source_encounter_id INT,
+                    concept_uuid CHAR(38),
+                    value_coded_uuid CHAR(38),
+                    value_datetime DATETIME,
+                    value_numeric DOUBLE,
+                    value_text TEXT
+                );
+            END $$
+            CREATE PROCEDURE migrate_tmp_obs()
+            BEGIN
+                INSERT INTO obs (
+                    obs_id, person_id, encounter_id, obs_group_id, obs_datetime, location_id, concept_id,
+                    value_coded, value_numeric, value_datetime, value_text, creator, date_created, voided, uuid
+                )
+                SELECT
+                    o.obs_id, p.person_id, e.encounter_id, o.obs_group_id, e.encounter_date, e.location_id, q.concept_id,
+                    a.concept_id, o.value_numeric, o.value_datetime, o.value_text, 1, e.date_created, 0, uuid()
+                FROM tmp_obs o
+                    JOIN       hivmigration_patients p ON o.source_patient_id = p.source_patient_id
+                    JOIN       hivmigration_encounters e ON o.source_encounter_id = e.source_encounter_id
+                    LEFT JOIN  concept q ON q.uuid = o.concept_uuid
+                    LEFT JOIN  concept a ON a.uuid = o.value_coded_uuid;
+            END $$
+            DELIMITER ;
+        ''')
+
```

Comment:
I would expect the above 2 stored procedures to be defined and executed outside of this particular job, so that they can just be available and used by any of the individual jobs that need them.  The behavior should be consistent whether we are migrating viral loads, med pickups, or any other obs.

Reply:
Yeah, agreed, just getting this hacked together right now